model/src/payouts/quanto.rs

```diff
@@ -0,0 +1,609 @@
+        calculate_initial_margin(
+            self.initial_price,
+            self.n_contracts,
+            self.leverage_short,
+            self.multiplier,
+        )
+        .context("Could not calculate short's initial margin")
+    }
+
+    /// Compute the profit and loss (PNL) at the given `closing_price`.
+    fn pnl_at_closing_price(&self, closing_price: u64) -> Result<Pnl> {
+        Pnl::new(
+            self.initial_price,
+            closing_price,
+            self.multiplier,
+            self.n_contracts,
+        )
+    }
+}
+
+/// Compute the initial BTC margin that a party has to put up, according to their `leverage`.
+fn calculate_initial_margin(
+    initial_price: u64,
+    n_contracts: u64,
+    leverage: Leverage,
+    multiplier: Decimal,
+) -> Result<Amount> {
+    let n_contracts = Decimal::from(n_contracts);
+    let leverage = Decimal::from(leverage.get());
+    let initial_price = Decimal::from(initial_price);
+
+    let margin = (n_contracts * initial_price * multiplier) / leverage;
+    let margin = margin.round_dp_with_strategy(8, RoundingStrategy::AwayFromZero);
+    let margin = margin.to_f64().context("Could not convert margin to f64")?;
+    let margin = Amount::from_btc(margin).context("Could not convert margin to bitcoin::Amount")?;
+
+    Ok(margin)
+}
+
+/// The profit and loss (PNL).
+///
+/// It is convenient to model the calculations for long and short together, because one party's gain
+/// is the other one's loss and vice-versa. That is, the absolute value of PNL will be the same,
+/// with only the sign changing between the two parties.
+struct Pnl(SignedAmount);
+
+impl Pnl {
+    /// Compute the PNL of the contract.
+    ///
+    /// Call `Self::new().long()` and `Self::new().short()` to access the PNL values for long and
+    /// short respectively.
+    fn new(
+        initial_price: u64,
+        closing_price: u64,
+        multiplier: Decimal,
+        n_contracts: u64,
+    ) -> Result<Self> {
+        let initial_price = Decimal::from(initial_price);
+        let closing_price = Decimal::from(closing_price);
+        let n_contracts = Decimal::from(n_contracts);
+
+        let pnl = (closing_price - initial_price) * multiplier * n_contracts;
+        let pnl = pnl.to_f64().context("Could not convert PNL to f64")?;
+        let pnl = SignedAmount::from_btc(pnl)
+            .context("Could not convert PNL to bitcoin::SignedAmount")?;
+
+        Ok(Self(pnl))
+    }
+
+    /// The profit and loss (PNL) from the perspective of the party going long.
+    fn long(&self) -> SignedAmount {
+        self.0
+    }
+
+    /// The profit and loss (PNL) from the perspective of the party going short.
+    fn short(&self) -> SignedAmount {
+        self.0 * -1
+    }
+}
+
+/// Closing prices at which parties get liquidated i.e. when they get a 0 payout.
+struct LiquidationThresholds {
+    inner: u64, // TODO: Find a better name!
+    initial_price: u64,
```

Comment:
What quantity does the u64 represent here? Sats?

Reply:
Are you asking about `initial_price`? If so, the units are whole USD, which is why we use a `u64`. 

The reason why I think using whole dollars makes sense is because the oracle only attests to the nearest dollar! So the extra precision seems unnecessary to me.

Reply:
That makes sense - I just wasn't sure what unit of measurement the u64 referred to